### PR TITLE
KafkaSource and binding v1beta1<>v1alpha1 conversion

### PR DESCRIPTION
**Pull request by aliok**
Part of #1332 

## Proposed Changes

- KafkaSource v1beta1<>v1alpha1 conversion test and conversion
- Same thing for KafkaBinding
- KafkaBinding and KafkaSource conversations share code

### CODE REVIEW

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion_test.go [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion_test.go:20]: Format Go code:
```suggestion
	"context"
	"testing"

```

File: kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion_test.go [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	"knative.dev/eventing-contrib/kafka/source/pkg/apis/bindings/v1alpha1"
+	v1beta1 "knative.dev/eventing-contrib/kafka/source/pkg/apis/sources/v1beta1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"testing"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/sources/v1alpha1/kafka_conversion_test.go:29]: Format Go code:
```suggestion

```

**Comment by aliok** [comment]
/hold
Forgot to test the other way around

File: kafka/source/pkg/apis/bindings/v1alpha1/kafka_conversion_test.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/bindings/v1alpha1/kafka_conversion_test.go:20]: Format Go code:
```suggestion
	"context"
	"testing"

```

File: kafka/source/pkg/apis/bindings/v1alpha1/kafka_conversion_test.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/tracker"
+	"testing"
```
Comment (mattmoor) [inline kafka/source/pkg/apis/bindings/v1alpha1/kafka_conversion_test.go:23]: Format Go code:
```suggestion

```

**Pull request merged**